src/main/java/com/jcraft/jsch/ChannelX11.java

```diff
@@ -45,8 +46,8 @@ class ChannelX11 extends Channel {
   static byte[] cookie = null;
   private static byte[] cookie_hex = null;
 
-  private static Hashtable<Session, byte[]> faked_cookie_pool = new Hashtable<>();
-  private static Hashtable<Session, byte[]> faked_cookie_hex_pool = new Hashtable<>();
+  private static final Map<Session, byte[]> faked_cookie_pool = new ConcurrentHashMap<>();
```

Comment:
I would just keep faked_cookie_pool a HashMap and maintain the `synchronized` on `faked_cookie_pool` that was removed anyway.